h2/connection.py

```diff
@@ -586,8 +593,11 @@ def initiate_upgrade_connection(self, settings_header=None):
             For servers, returns nothing.
         :rtype: ``bytes`` or ``None``
         """
-        frame_data = None
+        self.config.logger.debug(
+            "Upgrade connection.\nSettings frame: %s", self.local_settings
```

Comment:
Most log parsing/collecting/etc tools are fully capable of understanding multi-line log messages. Especially people with Python based infrastructure that may log an exception or traceback are familiar dealing with this. I'm not exonerating that in this particular case, but multi-line log messages aren't as bad as described here. (Spoken as someone who has dealt with 8 different Python services emitting multi-line log messages in 8 different ways + the different ways of each of their components).